alacritty_terminal/src/index.rs

```diff
@@ -44,33 +44,38 @@ impl<L> Point<L> {
 
     #[inline]
     #[must_use = "this returns the result of the operation, without modifying the original"]
-    pub fn sub(mut self, num_cols: usize, length: usize, absolute_indexing: bool) -> Point<L>
+    pub fn sub(mut self, num_cols: usize, rhs: usize, absolute_indexing: bool) -> Point<L>
     where
-        L: Copy + Add<usize, Output = L> + Sub<usize, Output = L>,
+        L: Copy + Default + Into<Line> + Add<usize, Output = L> + Sub<usize, Output = L>,
     {
-        let line_changes = f32::ceil(length.saturating_sub(self.col.0) as f32 / num_cols as f32);
+        let line_changes =
+            f32::ceil(rhs.saturating_sub(self.col.0) as f32 / num_cols as f32) as usize;
         if absolute_indexing {
-            self.line = self.line + line_changes as usize;
+            self.line = self.line + line_changes;
+        } else if self.line.into() > Line(line_changes) {
+            self.line = self.line - line_changes;
         } else {
-            self.line = self.line - line_changes as usize;
+            self.line = Default::default();
```

Comment:
Yeah, we don't want to underflow. Partially because that panics in debug mode and partially because that doesn't logically make sense. If you move a point down and it goes below the viewport, you don't want it to appear at the top.